### PR TITLE
Apply `beforeBreadcrumb` on native iOS crumbs

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Call Sentry Flutters `beforeBreadcrumb` on native iOS breadcrumbs.

## :bulb: Motivation and Context

Relates to #1903

## :green_heart: How did you test it?

Unit tests. Tested with example app.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

Since we sync contexts only on iOS, this would only work for iOS native breadcrumbs. Functionality on Android would stay as before.

### CODE REVIEW

**Comment by buenaflor** [comment]
I assume this is not only specific to iOS crumbs but to native crumbs in general?

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### Features
 
 - Use `recordHttpBreadcrumbs` to set iOS `enableNetworkBreadcrumbs` ([#1884](https://github.com/getsentry/sentry-dart/pull/1884))
+- Apply `beforeBreadcrumb` on native iOS crumbs ([#1914](https://github.com/getsentry/sentry-dart/pull/1914))
```
Comment (buenaflor) [inline CHANGELOG.md:14]: I think it should be `native breadcrumbs` since it's not only for iOS? correct me if I'm wrong

**Review by buenaflor — COMMENTED** [review]
Left a comment. Generally looks good to me but maybe @stefanosiano can do a quick sanity check seeing if we want to have this behaviour or if there are any reasons not to do it this way

File: CHANGELOG.md (hunk 1 above)
Comment (denrase) [reply CHANGELOG.md:14]: This does not work on Android afaik, only iOS/macOS

Comment (buenaflor) [reply CHANGELOG.md:14]: ah yes my bad, the contexts integration is cocoa specific, nvm

**Comment by krystofwoldrich** [comment]
I was thinking about doing the same in React Native. Makes sense.

With the new context sync in v8, this will also work for Android.

- https://github.com/getsentry/sentry-dart/issues/1556

**Review by buenaflor — APPROVED** [review]
as said, let's also do it for android on v8, you can also create a pr on the v8 branch for android @denrase

**Pull request merged**